assets/src/stories-editor/components/inserter/menu.js

```diff
@@ -52,6 +52,7 @@ import { addQueryArgs } from '@wordpress/url';
 import BlockPreview from '../block-preview';
 import BlockTypesList from '../block-types-list';
 import { ALLOWED_TOP_LEVEL_BLOCKS } from '../../constants';
+import { isBlockAllowedOnPage } from '../../helpers';
```

Comment:
Why can't the hook be used here?

Reply:
Because this is still a class component, not a functional one. See #2538 for explanation.

Reply:
I am just worried about bug coming from not using select correctly here and getting stale data.

Reply:
I'll take a stab at it 👍

Reply:
Isn't the benefit of hooks if that you put them everyone and mix and match?

Reply:
How that `isBlockAllowedOnPage` is only used in `Menu`, I think we should get rid of the helper and move the logic to `withSelect`

Reply:
That could work yeah 👍